root/etc/profile.d/discovery.sh

```diff
@@ -0,0 +1,8 @@
+# /etc/profile.d/discovery.sh - shell extensions for discovery
+
+# Changes to this file should also be reflected in the setup-extension systemd
+# service file, as systemd doesn't source this file
+
+export LD_LIBRARY_PATH=/opt/extension/lib
+export RUBYLIB=/opt/extension/lib/ruby
+export PATH=$PATH:/opt/extension/bin
```

Comment:
The EnvironmentFile is written by discovery-setup and a new service called discovery-autostart reads that file and runs the script(s). I've also renamed discover-host to discovery-register just to be consistent, as it's a better description of the function.